logging/performance_reliability/logging-flow-control-mechanisms.adoc

```diff
@@ -0,0 +1,43 @@
+:_mod-docs-content-type: ASSEMBLY
+include::_attributes/common-attributes.adoc[]
+include::_attributes/attributes-openshift-dedicated.adoc[]
+[id="logging-flow-control-mechanisms"]
+= Flow control mechanisms
+:context: logging-flow-control-mechanisms
+
+toc::[]
+
+If logs are being produced faster than they can be collected, it can be difficult to predict or control the volume of logs to an output.
+
+This can result in logs being lost, or in the event of a system outage, when log buffers build up without user control, this can cause long recovery times and high latency when the connection is restored.
+
+As an administrator, you can limit logging rates by configuring flow control mechanisms for your {logging}.
+
+[id="logging-configuring-flow-control-benefits"]
+== Benefits of flow control mechanisms
+
+* The cost and volume of logging can be predicted more accurately in advance.
+* Noisy containers cannot produce unbounded log traffic that drowns out other containers.
+* Ignoring logs reduces the load on the logging infrastructure.
+* High-value logs can be preferred over low-value logs by assigning higher rate limits.
+
+[id="logging-configuring-flow-control-about-rate-limits"]
+== Configuring rate limits
+
+Rate limits are configured per collector, which means that the total amount of collected logs is the number of collector instances multiplied by the rate limit. For example, if you have 3 collector instances, and a maximum rate limit of 10 records per second set for each collector, the total amount of collected logs is a maximum of 30 records per second.
+
+Rate limits are set in number of records instead of bytes. This is because the exact byte size of a record as written to an output can vary, due to transformations, different encodings, or other factors.
+
+You can configure rate limits in the `ClusterLogForwarder` custom resource (CR) in two ways:
+
+Output rate limit:: You can limit the rate of outbound logs to match the network and storage capacity of selected outputs. The output rate limit controls the aggregated per-output rate.
+
+Input rate limit:: You can limit the rate of log collection for selected containers. The input rate limit controls per-container log collection throttling.
```

Comment:
```suggestion
Input rate limit:: You can limit the per-container rate of log collection for selected containers.
```

Reply:
"throttling" is a perfectly good word to use in this context but it is a synonym for "flow control" and "rate limiting", so it is redundant. We don't use "throttling" elsewhere so it pops out more than it should.